connectivity/pxp_reporter_makefile/build/Makefile

```diff
@@ -0,0 +1,397 @@
+
+SECONDARY_FLASH += \
+$(BUILD_ARTIFACT_NAME).bin \
+
+SECONDARY_SIZE += \
+$(BUILD_ARTIFACT_NAME).siz \
+
+# All Target
+ifeq (ram, $(word 2,$(subst _, ,$(BUILD_CONFIG))))
+all: pre-build main-build
+else
+# For qspi builds, additional post-build is added to create a .img with the product and image headers
+all: pre-build main-build post-build
+endif
+
+# Main-build Target
+main-build: check-build-config $(BUILD_ARTIFACT) secondary-outputs 
+ 
+# Tool invocations
+
+# Generate .elf 
+$(BUILD_ARTIFACT): $(OBJS) $(USER_OBJS) $(OPTIONAL_TOOL_DEPS)
+	@echo 'Building target: $@'
+	@echo 'Invoking: Cross ARM C Linker'
+	@$(CC) $(LDFLAGS) \
+	 -Wl,-Map,"$(BUILD_OUTPUT_DIR)/$(BUILD_ARTIFACT_NAME).map" \
+	--specs=nano.specs --specs=nosys.specs -o "$(BUILD_OUTPUT_DIR)/$@" $(OBJS) $(USER_OBJS) $(LIBS)
+	@echo 'Finished building target: $@'
+	@echo ' '
+
+# Generate .bin 
+$(SECONDARY_FLASH): $(BUILD_ARTIFACT) $(OPTIONAL_TOOL_DEPS)
+	@echo 'Invoking: Cross ARM GNU Create Flash Image'
+	arm-none-eabi-objcopy -O binary "$(BUILD_OUTPUT_DIR)/$<"  "$(BUILD_OUTPUT_DIR)/$@" 
+	@echo 'Finished building: $@'
+	@echo ' '
+
+# Print size
+$(SECONDARY_SIZE): $(BUILD_ARTIFACT) $(OPTIONAL_TOOL_DEPS)
+	@echo 'Invoking: Cross ARM GNU Print Size'
+	@arm-none-eabi-size --format=berkeley "$(BUILD_OUTPUT_DIR)/$<"
+	@echo 'Finished building: $@'
+	@echo ' '
+
+# make clean will clean ALL build configurations {debug_qpsi, debug_qpsi_suota, debug_ram, release_qspi, release_qspi_suota, release_ram}
+# setting BUILD_CONFIG when calling clean will clean only that configuration
+# For example:
+# make BUILD_CONFIG=debug_qspi clean 
+# will clean only the debug_qspi build
+clean:
+ifeq (,$(BUILD_CONFIG))
+	@echo 'Cleaning all build configuration outputs'
+	@echo ' '
+	$(foreach config,$(BUILD_CONFIG_TYPES),$(call clean_template,$(config)))
+	@echo 'Finished cleaning all build configurations'
+	@echo ' '
+
+else
+	@echo 'Cleaning $(BUILD_CONFIG) build outputs'
+	$(call clean_extensions_template, $(BUILD_CONFIG))
+	-@$(RM) $(OBJS) $(SECONDARY_FLASH) $(SECONDARY_SIZE) $(ASM_DEPS) $(S_UPPER_DEPS) $(C_DEPS)
+	@echo 'Finished cleaning $(BUILD_CONFIG)'
+	@echo ' '
+
+endif
+
+# Check that a valid build configuration has been selected
+check-build-config:
+ifeq (,$(findstring $(BUILD_CONFIG), $(BUILD_CONFIG_TYPES)))
+	$(error Invalid build configuration. Please set BUILD_CONFIG to one of {$(BUILD_CONFIG_TYPES)})
+endif
+
+# Generate linker scripts
+pre-build: check-build-config
+	-@echo 'Generate linker scripts.'
+	-@mkdir -p $(BUILD_OUTPUT_DIR)
+	@make generate_ldscripts DEVICE=$(DEVICE) PRE_BUILD_EXTRA_DEFS="" \
+	APP_CONFIG_H=$(APP_CONFIG_H) \
+	CC=$(CC) \
+	BSP_CONFIG_DIR="$(SDKROOT)/bsp/config" \
+	MIDDLEWARE_CONFIG_DIR="$(SDKROOT)/middleware/config" \
+	LDSCRIPT_PATH="$(SDKROOT)/bsp/ldscripts/ble_projects"
+	-@echo ' '
+			
+# For qspi builds, create a .img with product and image header. 
+post-build: main-build
```

Comment:
The post-build is assuming that the file "program_qspi.xml" exists and this is not the case with an OoB SDK. Let's add a comment explaining this or we could make the post-build optional

Reply:
Updated in [1ba3fc9](https://github.com/dialog-semiconductor/BLE_SDK10_examples/pull/30/commits/1ba3fc9295fb8640ee6ef1eb723ecbc2f42c2ad1)

Made post-build optional and added a note in the README explaining the dependency on program_qspi.xml and how it can be created.